homeassistant/components/light/lw12wifi.py

```diff
@@ -0,0 +1,173 @@
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.lw12wifi/
+"""
+import logging
+import time
+
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_EFFECT, ATTR_HS_COLOR, ATTR_TRANSITION,
+    PLATFORM_SCHEMA, SUPPORT_BRIGHTNESS, SUPPORT_COLOR, SUPPORT_EFFECT,
+    SUPPORT_TRANSITION, Light)
+from homeassistant.const import (
+    CONF_BRIGHTNESS, CONF_EFFECT, CONF_HOST, CONF_NAME, CONF_PORT, CONF_RGB)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.color as color_util
+
+REQUIREMENTS = ['lw12==0.9.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TRANSITION = 'transition'
+
+DEFAULT_BRIGHTNESS = 255
+DEFAULT_EFFECT = None
+DEFAULT_NAME = 'LW-12 FC'
+DEFAULT_PORT = 5000
+DEFAULT_RGB = [255, 255, 255]
+DEFAULT_TRANSITION = 128
+
+SUPPORT_LW12 = (
+        SUPPORT_BRIGHTNESS | SUPPORT_EFFECT | SUPPORT_COLOR |
+        SUPPORT_TRANSITION
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_BRIGHTNESS, default=DEFAULT_BRIGHTNESS):
+        vol.All(vol.Coerce(int), vol.Range(min=0, max=255)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_RGB, default=DEFAULT_RGB):
+        vol.All(list, vol.Length(min=3, max=3),
+                [vol.All(vol.Coerce(int), vol.Range(min=0, max=255))]),
+    vol.Optional(CONF_TRANSITION, default=DEFAULT_TRANSITION):
+        vol.All(vol.Coerce(int), vol.Range(min=0, max=255)),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up LW-12 WiFi LED Controller platform."""
+    import lw12
+
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    rgb = config.get(CONF_RGB)
+    brightness = config.get(CONF_BRIGHTNESS)
+    effect = config.get(CONF_EFFECT)
+    transition = config.get(CONF_TRANSITION)
+
+    light = lw12.LW12Controller(host, port)
+
+    add_devices([LW12WiFi(
+        name, light, rgb, brightness, effect, transition)])
+
+
+class LW12WiFi(Light):
+    """Representation of a LW-12 WiFi LED Controller."""
+
+    def __init__(self, name, light, rgb_color, brightness, effect, transition):
+        """Initialisation of LW-12 WiFi LED Controller."""
+        self._light = light
+        self._name = name
+        self._rgb_color = rgb_color
+        self._brightness = brightness
+        self._state = None
+        self._effect = effect
+        self._transition_speed = transition
+
+    @property
+    def name(self):
+        """Return the display name of the controlled light."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def rgb_color(self):
```

Comment:
`rgb_color` is not used by the light component base. This property can be removed.